servicemanager/smprocess.py

```diff
@@ -176,7 +172,7 @@ def __init__(self, ppid, pid, uptime, mem, args):
     def processes_matching(regex, processes=None):
         processes = processes or SmProcess.all_processes()
```

Comment:
could this be called higher in the stack? Thus meaning it is only called once per "sm --status"?